warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormat.java

```diff
@@ -0,0 +1,137 @@
+     * a given Key spans more index blocks.
+     *
+     * @param config
+     * @param fileSplit
+     * @return
+     * @throws IOException
+     */
+    public static List<InputSplit> getSplits(Configuration config, FileSplit fileSplit) throws IOException {
+        log.info("getting splits for: " + fileSplit);
+        Path rfile = fileSplit.getPath();
+        FileSystem fs = rfile.getFileSystem(config);
+
+        if (!fs.exists(rfile)) {
+            throw new IllegalArgumentException(rfile + " does not exist");
+        }
+
+        CryptoService cs = CryptoFactoryLoader.getServiceForClient(CryptoEnvironment.Scope.TABLE, config.getPropsWithPrefix(TABLE_CRYPTO_PREFIX.name()));
+        CachableBlockFile.CachableBuilder cb = new CachableBlockFile.CachableBuilder().fsPath(fs, rfile).conf(config).cryptoService(cs);
+        RFile.Reader rfileReader = new RFile.Reader(cb);
+
+        // get the first and last keys to bound the blocks while creating splits
+        Key firstKey = rfileReader.getFirstKey();
+        Key lastKey = rfileReader.getLastKey();
+
+        // use the index blocks to create the splits
+        FileSKVIterator iter = rfileReader.getIndex();
+
+        // track the last split key to since multiple splits with the same split key MUST be in the same block
+        Key lastSplit = firstKey;
+        List<InputSplit> splits = new ArrayList<>();
+        int blkCount = 0;
+        int splitBlocks = 0;
+
+        int minBlocksPerSplit = config.getInt(MIN_BLOCKS_PER_SPLIT, 1);
+        log.info("Blocks per split: " + minBlocksPerSplit);
+
+        Key top = null;
+        while (iter.hasTop()) {
+            splitBlocks++;
+            top = iter.getTopKey();
+
+            if (!top.equals(lastSplit) && splitBlocks >= minBlocksPerSplit) {
+                splits.add(new RFileSplit(fileSplit.getPath(), fileSplit.getStart(), fileSplit.getLength(), fileSplit.getLocations(), blkCount, splitBlocks,
+                                top));
+                blkCount += splitBlocks;
+                lastSplit = iter.getTopKey();
+                splitBlocks = 0;
+            }
+            iter.next();
+        }
+
+        // if there is a gap between the last index key top key and the lastKey in the file or there hasn't been a split created yet
+        if (top.equals(lastKey) || splits.isEmpty()) {
+            // add the last split
+            splits.add(new RFileSplit(fileSplit.getPath(), fileSplit.getStart(), fileSplit.getLength(), fileSplit.getLocations(), blkCount, splitBlocks + 1,
+                            top));
+        }
+
+        return splits;
+    }
+
+    // get the splits for the rfile
+    public static List<InputSplit> getSplits(Configuration config, Path rfile) throws IOException {
+        return getSplits(config, new FileSplit(rfile, 0, 0, new String[0]));
```

Comment:
It looks like this method is only used by unit tests and it's a suboptimal split since it's stubbing out the length and the locations. Thoughts on moving it to the unit test as a utility method there, just to prevent anyone from using this in production code.

Reply:
agree